### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by naoyam**
Had a sanitizer error with a new test in the IterDomain resize branch. Turned out there's this obvious bug. A bit scary as the impact could be very broad.

### CODE REVIEW

**Comment by naoyam** [comment]
Another note: the analysis of the parallel dimension map can be significantly simplified with the new expr simplifier.

**Review by zasdfgbnm — APPROVED** [review]
Agree that expr simplifier will be a better approach, but don't have to be this PR

File: third_party/nvfuser/csrc/parallel_dimension_map.cpp [hunk 1]
```diff
@@ -303,7 +303,7 @@ bool ParallelDimensionMap::equalDim(Val* dim1, Val* dim2) {
         dim2_def->as<UnaryOp>()->getUnaryOpType()))) {
     for (const auto i : c10::irange(dim1_def->inputs().size())) {
       (void)i; // Suppress unused variable warning
```
Comment (csarofeen) [inline third_party/nvfuser/csrc/parallel_dimension_map.cpp:305]: Can't this line be removed now?

Comment (naoyam) [reply third_party/nvfuser/csrc/parallel_dimension_map.cpp:305]: I still see compilation errors with unused variables, so no I don't think so.

Comment (csarofeen) [reply third_party/nvfuser/csrc/parallel_dimension_map.cpp:305]: It makes sense there was one before... but you're using `i` now... that's really strange, no?

Comment (naoyam) [reply third_party/nvfuser/csrc/parallel_dimension_map.cpp:305]: Aha!

Comment (naoyam) [reply third_party/nvfuser/csrc/parallel_dimension_map.cpp:305]: #2520

**Pull request merged**